src/slack_interaction_handler.ts

```diff
@@ -197,10 +198,19 @@ export async function handleVoterStatusUpdate({
     } else {
       // Take the blocks and replace the initial_option with the new status, so that
       // even when Slack is refreshed this new status is shown.
-      SlackBlockUtil.populateDropdownNewInitialValue(
-        payload.message.blocks,
-        selectedVoterStatus as VoterStatus
-      );
+      if (
+        !SlackBlockUtil.populateDropdownNewInitialValue(
+          payload.message.blocks,
+          payload.actions
+            ? payload.actions[0].action_id
+            : SlackActionId.VOTER_STATUS_DROPDOWN,
```

Comment:
I ran into issues when testing because messages created before we started using static action IDs stopped working. Those old blocks do have Slack-auto-generated `action_id`s and they should be specified in the payload though (the type says `payload.actions` is sometimes undefined, hence this ternary), so we can try to use those `action_id`s in the payload if available.

Reply:
Sounds good, this makes sense. If `payload.actions` is sometimes undefined, do you know if it's at least consistent?

Reply:
Yup, it's consistent.